### PR TITLE
Add experimental support for default slot values

### PR DESCRIPTION
**Pull request by joelhawksley**
### What are you trying to accomplish?

This PR aims to close #1861 and #614, implementing an experimental approach for default slot values.

### What approach did you choose and why?

I added an experimental module that must be included to use the feature, adopting the API suggested by @BlakeWilliams in https://github.com/ViewComponent/view_component/pull/1861#issuecomment-1785454288.

### Anything you want to highlight for special attention from reviewers?

Do we want to allow default value(s) for `renders_many` slots? I don't think we need to solve it now, but something to consider adding later.

### CODE REVIEW

**Comment by reeganviljoen** [comment]
@joelhawksley would you mind adding docs about this, as I feel users would genuinely like this ❤️

File: test/sandbox/test/slotable_test.rb [hunk 1]
```diff
@@ -764,4 +764,16 @@ def test_forwarded_slot_renders_correctly
 
     assert_text "Target content", count: 1
   end
+
+  def test_slotable_default
```
Comment (Spone) [inline test/sandbox/test/slotable_test.rb:768]: Shouldn't we test that specifying the slot correctly overrides the default?

**Review by Spone — CHANGES_REQUESTED** [review]
Great to see this addition! Nice API :)

**Comment by joelhawksley** [comment]
@reeganviljoen done, let me know what you think! https://github.com/ViewComponent/view_component/pull/2063/files#diff-867de23963ba79a34d995f1d16db1d42da92c63d86481bf59031a7d0ebc6f837

**Review by reeganviljoen — APPROVED** [review]
@joelhawksley LGTM🚀

**Review by Spone — APPROVED** [review]
👌

**Pull request merged**